### PR TITLE
[Data] Add Huggingface Image Classification Batch Inference Example

### PR DESCRIPTION
**Pull request by amogkam**
Huggingface example for image classification batch inference.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by raulchen** [comment]
> {class}`Ray Dataset <ray.data.dataset.Dataset>`

can you check if this link is correct? IIRC, it should be `ray.data.Dataset`

> [Ray Data](https://github.com/ray-project/ray/blob/545cffe94bb25c6c3df8e9b25360c1bf2a01e3d2/doc/source/data/examples/data) also supports reading from a variety of other [datasources and formats](https://github.com/ray-project/ray/blob/545cffe94bb25c6c3df8e9b25360c1bf2a01e3d2/doc/source/data/examples/loading_data).

These 2 links are not correct.

> Inspecting the schema, we can see that there is 1 columns in the dataset containing the images stored as Numpy arrays.

"columns" -> "column"

> There are a couple unique properties about GPU inference:
> 1. Model initialization is usually pretty expensive

Model initialization is not unique to GPU inference. Better change this to something like "There are a couple facts that impact the performance of GPU inference" to avoid confusion.

> The actual inference model inference logic is in the __call__ method, which will be called for each batch.

Drop the first "inference"? 

> Since we are using a class in this case, Function-based the UDF will run as long-running [Ray actors](https://docs.ray.io/en/latest/ray-core/key-concepts.html#actors).

Should be "class-based" here?

**Review by raulchen — APPROVED** [review]
LGTM. Only a few small comments.

**Comment by amogkam** [comment]
Made the necessary rewordings. Not sure what's wrong the links though, those are working fine for me?

File: .buildkite/pipeline.gpu_large.yml [hunk 1]
```diff
@@ -48,6 +48,10 @@
     - DOC_TESTING=1 TRAIN_TESTING=1 TUNE_TESTING=1 ./ci/env/install-dependencies.sh
     - pip install -Ur ./python/requirements/ml/requirements_ml_docker.txt
     - ./ci/env/env_info.sh
+    # Test examples with newer version of `transformers`
+    # TODO(amogkam): Remove when https://github.com/ray-project/ray/issues/36011 
+    # is resolved.
+    - pip install -U transformers
```
Comment (amogkam) [inline .buildkite/pipeline.gpu_large.yml:54]: fyi @krfricke adding this until we can make the upgrade in the requirements files.

**Pull request merged**